src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,148 @@
+package duke;
+
+import java.util.zip.DataFormatException;
+
+import duke.command.AddCommand;
+import duke.command.ByeCommand;
+import duke.command.Command;
+import duke.command.DeleteCommand;
+import duke.command.DoneCommand;
+import duke.command.ListCommand;
+
+
+/**
+ * Understands inputs given by user and validates them.
+ */
+public class Parser {
+
+    /**
+     * Understands input from user
+     *
+     * @param userText input from user
+     * @param taskList list of tasks user has currently
+     * @return boolean value of whether conversation is to be continued
+     */
+    public static boolean understandText(String userText, TaskList taskList) {
+        try {
+            String edittedAnswer = userText.strip().toLowerCase();
+            String[] answers = userText.split(" ");
+            if (answers.length == 2 && answers[0].equals(Commands.DONE.text)) {
+                Parser.furtherUnderstandTaskNumber(answers[0], answers[1], taskList);
+
+            } else if (answers.length == 2 && answers[0].equals(Commands.DELETE.text)) {
+                Command command = new DeleteCommand();
+                Parser.furtherUnderstandTaskNumber(answers[0], answers[1], taskList);
+
+            } else if (edittedAnswer.equals(Commands.LIST.text)) {
+                Command command = new ListCommand();
+                command.execute(" ", taskList);
+
+            } else if (edittedAnswer.equals(Commands.EXIT.text)) {
+                Command command = new ByeCommand();
+                command.execute(" ", taskList);
+                return false;
+
+            } else if (answers[0].equals(Commands.TODO.text)
+                    || answers[0].equals(Commands.DEADLINE.text)
+                    || answers[0].equals(Commands.EVENT.text)) {
+                Parser.furtherUnderstandTask(userText, taskList);
+
+            } else {
+                throw new DukeCannotUnderstandException();
+
+            }
+        } catch (DukeCannotUnderstandException e) {
+            System.out.println(e.getMessage() + "\n" + Ui.LINE);
+        }
+        return true;
+    }
+
+    private static void furtherUnderstandTaskNumber(String stringCommand, String answer, TaskList taskList) {
```

Comment:
Instead of using "further", perhaps this can method can be simply named as understandTaskNumber()?

Reply:
Right, thank you for the suggestions, very valid, will take note and change.